rust/connlib/tunnel/src/client.rs

```diff
@@ -789,6 +824,10 @@ impl ClientState {
             .chain(self.dns_mapping.left_values().copied().map(Into::into))
     }
 
+    fn is_resource_enabled(&self, resource: &ResourceId) -> bool {
```

Comment:
Should this check that we know the resource ID at all?

Reply:
This does it since if it's a DNS resource it should be in that map and if it's not we never hit this code-path

Reply:
It is a defense-in-depth thing and I am pretty confident in our tests so it may not be necessary. It is more of a nit because just from reading the fn name, it is not clear that this function will return you `true` for resources that don't exist which I'd say is a surprising return value.